scripts/ci/rust_checks.py

```diff
@@ -197,44 +214,69 @@ def sdk_variations(results: list[Result]) -> None:
+    "aarch64-apple-darwin",
+    "wasm32-unknown-unknown",
+    "x86_64-apple-darwin",
+    "x86_64-pc-windows-gnu",
+    "x86_64-pc-windows-msvc",
+    "x86_64-unknown-linux-gnu",
+    "x86_64-unknown-linux-musl",
+]
+
+
 def cargo_deny(results: list[Result]) -> None:
     # Note: running just `cargo deny check` without a `--target` can result in
     # false positives due to https://github.com/EmbarkStudios/cargo-deny/issues/324
     # Installing is quite quick if it's already installed.
     results.append(run_cargo("install", "--locked cargo-deny@^0.17"))
-    results.append(
-        run_cargo("deny", "--all-features --exclude-dev --log-level error --target aarch64-apple-darwin check"),
-    )
-    results.append(
-        run_cargo("deny", "--all-features --exclude-dev --log-level error --target i686-pc-windows-gnu check"),
-    )
-    results.append(
-        run_cargo("deny", "--all-features --exclude-dev --log-level error --target i686-pc-windows-msvc check"),
-    )
-    results.append(
-        run_cargo("deny", "--all-features --exclude-dev --log-level error --target i686-unknown-linux-gnu check"),
-    )
-    results.append(
-        run_cargo("deny", "--all-features --exclude-dev --log-level error --target wasm32-unknown-unknown check"),
-    )
-    results.append(
-        run_cargo("deny", "--all-features --exclude-dev --log-level error --target x86_64-apple-darwin check"),
-    )
-    results.append(
-        run_cargo("deny", "--all-features --exclude-dev --log-level error --target x86_64-pc-windows-gnu check"),
-    )
-    results.append(
-        run_cargo("deny", "--all-features --exclude-dev --log-level error --target x86_64-pc-windows-msvc check"),
-    )
-    results.append(
-        run_cargo("deny", "--all-features --exclude-dev --log-level error --target x86_64-unknown-linux-gnu check"),
-    )
-    results.append(
-        run_cargo("deny", "--all-features --exclude-dev --log-level error --target x86_64-unknown-linux-musl check"),
-    )
-    results.append(
-        run_cargo("deny", "--all-features --exclude-dev --log-level error --target x86_64-unknown-redox check"),
-    )
+
+    for target in deny_targets:
+        results.append(
+            run_cargo("deny", f"--all-features --exclude-dev --log-level error --target {target} check"),
+        )
+
+
+def denied_sdk_deps(results: list[Result]) -> None:
+    """
+    Check for disallowed SDK dependencies.
+
+    This protects against leaking UI dependencies into the SDK.
+    """
+
+    # Installing is quite quick if it's already installed.
+    results.append(run_cargo("install", "--locked cargo-tree@^0.29.0"))
+
+    # Sampling of dependencies that should never show up in the SDK, unless the viewer is enabled.
+    disallowed_dependencies = [
+        "re_viewer",
+        "wgpu",
+        "egui",
+        "winit",
+        "rfd",  # File dialog library.
+        "objc2-ui-kit",  # MacOS system libraries.
+        "wayland-sys",  # Linux windowing.
```

Comment:
We could add "cocoa" to this list: https://crates.io/crates/cocoa

Also sort the list :)

Reply:
> We could add "cocoa" to this list: [crates.io/crates/cocoa](https://crates.io/crates/cocoa)

sure why not, but it actually doesn't show up in our cargo.lock 🤷 

The sorting of the list is rather deliberate from "big to small" - `re_viewer` will obviously draw in `rfd`, so if `re_viewer` leaked in I want an error about that rather than `rfd`.
I figure the better way to go about that might be to collect all disallowed crates 🤔

Reply:
Ah, that ordering makes perfect sense! Please add a comment to that effect :)